### PR TITLE
fix: Always prompt if multiple MFA methods are running

### PR DESCRIPTION
**Pull request by codingllama**
Fixes a scenario where the MFA prompt wouldn't show if the user has both OTP and WebAuthn devices, but no WebAuthn device is plugged.

I've moved the prompt for "dual prompt" scenarios (OTP+WebAuthn) outside of `wancli` and directly into `lib/client/mfa`. This prompts earlier, but is overall more resilient to some goroutine failing (which was the case here).

I also fixed a (theoretical?) deadlock on the "WebAuthn with PIN" scenario (`otpDone` is never closed in that branch). I can't quite make it happen with the devices I have today (MFA scenarios don't prompt for PIN and passkeys go to other branches), but that doesn't mean it couldn't happen with some other/future device.

#43072

Changelog: Fix missing tsh MFA prompt in certain OTP+WebAuthn scenarios

### CODE REVIEW

**Comment by codingllama** [comment]
Backport to v14 is unlikely to happen, but I'll give it a shot when we get there.

**Comment by codingllama** [comment]
Rebased and resolved conflicts. No noteworthy changes.

**Comment by codingllama** [comment]
I've tested various combinations manually to make sure that it works and no other prompts are affected (they shouldn't be, but no harm in testing). I've also tested tsh on Windows.

Scenarios include:

- OTP + WebAuthn login (both outcomes, no security key plugged) -> this is the bug scenario
- OTP + WebAuthn login (both outcomes)
- OTP only login
- Webauthn only login
- Passkey login

Plus the same matrix but on Windows.

**Review by zmb3 — APPROVED** [review]

File: lib/client/mfa.go [hunk 1]
```diff
@@ -51,8 +49,9 @@ func (tc *TeleportClient) createAuthenticateChallenge(ctx context.Context, req *
 	return rootClient.CreateAuthenticateChallenge(ctx, req)
 }
 
-// WebauthnLoginFunc matches the signature of [wancli.Login].
-type WebauthnLoginFunc func(ctx context.Context, origin string, assertion *wantypes.CredentialAssertion, prompt wancli.LoginPrompt, opts *wancli.LoginOpts) (*proto.MFAAuthenticateResponse, string, error)
+// WebauthnLoginFunc is a function that performs WebAuthn login.
+// Mimics the signature of [webauthncli.Login].
+type WebauthnLoginFunc = libmfa.WebauthnLoginFunc
```
[the remark this answers is not in the document]
Comment (codingllama) [reply lib/client/mfa.go:54]: The type alias is fine here, I think. I would rather have l/c/mfa.WebauthnLoginFunc everywhere, but I'm afraid the larger refactor would make backports more difficult.

**Comment by codingllama** [comment]
Friendly ping @Joerger @marcoandredinis @probakowski ?

**Review by marcoandredinis — APPROVED** [review]
Changes look good
I've also tested this locally and it is now asking for the OTP on my use case as well.
Thank you for the quick fix

**Pull request merged**